app/services/broadcasts/welcome_notification/introduction.rb

```diff
@@ -1,37 +1,36 @@
-# Generates a broadcast to be delivered as a notification.
 module Broadcasts
   module WelcomeNotification
-    class Generator
-      def initialize(receiver_id)
-        @user = User.find(receiver_id)
+    class Introduction
+      attr_reader :user
+
+      def initialize(user)
+        @user = user
       end
 
-      def self.call(*args)
-        new(*args).call
+      def self.send(*args)
+        new(*args).send
       end
 
-      def call
+      def send
         return if commented_on_welcome_thread? || received_notification?
 
         Notification.send_welcome_notification(user.id, welcome_broadcast.id)
       end
 
-      def received_notification?
-        Notification.exists?(notifiable: welcome_broadcast, user: user)
-      end
+      private
 
       def commented_on_welcome_thread?
         welcome_thread = Article.admin_published_with("welcome").first
         Comment.where(commentable: welcome_thread, user: user).any?
       end
 
-      private
+      def received_notification?
```

Comment:
good call on making this `private`! i must have missed that it wasn't when i last reviewed this file